collector/systemd_linux.go

```diff
@@ -500,9 +500,9 @@ func getSystemdVersion(logger log.Logger) int {
 		level.Warn(logger).Log("msg", "Unable to get systemd version property, defaulting to 0")
 		return 0
 	}
-	re := regexp.MustCompile(`[0-9][0-9][0-9]`)
+	re := regexp.MustCompile(`[0-9][0-9][0-9](\.[0-9])?`)
```

Comment:
Should we make this handle more digits?

```suggestion
	re := regexp.MustCompile(`[0-9][0-9][0-9](\.[0-9]+)?`)
```

Reply:
I think so, yes:
https://github.com/systemd/systemd-stable/tags

The only thing here we can't represent is the -rc versions

Reply:
I have a couple refactoring ideas.